### ISSUE TITLE
Support for Nordictrack S27i bike (iFit2)

### ISSUE THREAD

**Issue by victorypoint** [cagnulein/QZCompanionNordictrackTreadmill#121]
Please add support for S27i bike metrics using new iFit2 wolflog format as per C2950 treadmill support. Currently testing with 'TDF Companion IP' in QZ, and 'S22i Bike' in QZ Companion. Only incline is currently working. See attached wolflog for recorded metrics. Example metrics below:

```
13:04:41.675 INFO SDS Changed INCLINE from 0.0 % to 0.5 %
12:55:39.005 INFO SDS Changed RESISTANCE from 0.0 resistance to 1.0 resistance
12:55:39.007 INFO SDS Changed GEAR from 0.0 gear to 1.0 gear (appears to always be same values as resistance)
13:04:41.996 INFO SDS Changed KPH from 4.8 kph to 6.04 kph
```

Also please advise on recommended QZ bike settings required to work with Zwift - e.g. offsets, gains, toggles. 

[log.latest.txt](https://github.com/user-attachments/files/17691932/log.latest.txt)

**Comment by victorypoint** [comment]
Adding previous iFit1 wolflog metrics format in case you need it. Example recorded metrics:

```
20:16:10.8628 [Trace:FitPro] Changed Watts to: 4
20:16:10.8706 [Trace:FitPro] Changed RPM to: 15
20:16:12.7307 [Trace:FitPro] Changed Resistance to: 1
20:16:12.7407 [Trace:FitPro] Changed CurrentGear to: 1
20:16:12.7421 [Trace:FitPro] Changed MaxGear to: 24
20:16:12.7421 [Trace:FitPro] Changed GearOption to: 0
```

[2024-10-10_logs.txt](https://github.com/user-attachments/files/17691973/2024-10-10_logs.txt)

**Comment by cagnulein** [comment]
hi @victorypoint could you please add a debug log from qz too? thanks

**Comment by cagnulein** [comment]
argh @victorypoint i can't see the wattage in the new ifit2. do you see it? it the most important value for zwift
Let me know

**Comment by victorypoint** [comment]
> argh @victorypoint i can't see the wattage in the new ifit2. do you see it? it the most important value for zwift Let me know

@cagnulein, no unfortunately it's not recorded in that log. I did a longer workout and inspected all the logs. Still no watts in the glassos log but I found watts in another log at '/data/com.ifit.rivendell/files/.valinorlogs/log.latest.txt' Unfortunately watts is only reported every 10 minutes in this log.

11:51:43.520 INFO CDD Distance stat updated from 54.26 to 29.34
11:51:43.527 INFO CDD Calories stat updated from 5057.0 to 2691
11:51:43.530 INFO CDD Elevation stat updated from 723.0 to 344
11:51:43.532 INFO CDD Pace stat updated from 15.266666666666667 to 11:43
11:51:43.534 INFO CDD Watts stat updated from 147.0 to 29.34
12:01:29.249 INFO CDD Distance stat updated from 54.26 to 32.78
12:01:29.265 INFO CDD Calories stat updated from 5057.0 to 2779
12:01:29.272 INFO CDD Elevation stat updated from 723.0 to 383
12:01:29.283 INFO CDD Pace stat updated from 15.266666666666667 to 10:46
12:01:29.293 INFO CDD Watts stat updated from 147.0 to 32.78

Watts is displayed in real-time on the manual workout screen so I don't know why it's not recorded real-time as well. Very frustrating. Logs attached.

[glassos - log.latest.txt](https://github.com/user-attachments/files/17705673/glassos.-.log.latest.txt)
[mithlond - log.latest.txt](https://github.com/user-attachments/files/17705677/mithlond.-.log.latest.txt)
